src/biokbase/narrative/jobs/job.py

```diff
@@ -193,52 +212,82 @@ def __getattr__(self, name):
             "cell_id": lambda: self._acc_state.get("job_input", {})
             .get("narrative_cell_info", {})
             .get("cell_id", JOB_ATTR_DEFAULTS["cell_id"]),
-            "child_jobs": lambda: copy.deepcopy(
+            "job_id": lambda: self._acc_state.get("job_id"),
+            "params": lambda: copy.deepcopy(
+                self._acc_state.get("job_input", {}).get(
+                    "params", JOB_ATTR_DEFAULTS["params"]
+                )
+            ),
+            "retry_parent": lambda: self._acc_state.get(
+                "retry_parent", JOB_ATTR_DEFAULTS["retry_parent"]
+            ),
+            "run_id": lambda: self._acc_state.get("job_input", {})
+            .get("narrative_cell_info", {})
+            .get("run_id", JOB_ATTR_DEFAULTS["run_id"]),
+            "tag": lambda: self._acc_state.get("job_input", {})
+            .get("narrative_cell_info", {})
+            .get("tag", JOB_ATTR_DEFAULTS["tag"]),
+            "user": lambda: self._acc_state.get("user", JOB_ATTR_DEFAULTS["user"]),
+            "wsid": lambda: self._acc_state.get("wsid", JOB_ATTR_DEFAULTS["wsid"]),
+            # the following properties can change whilst a job is in progress
+            # TODO: add the status attribute!
+            "child_jobs": lambda force_refresh=True: copy.deepcopy(
                 # TODO
                 # Only batch container jobs have a child_jobs field
                 # and need the state refresh.
                 # But KBParallel/KB Batch App jobs may not have the
                 # batch_job field
-                self.refresh_state(force_refresh=True).get(
+                self.refresh_state(force_refresh=force_refresh).get(
```

Comment:
this is deeply annoying to me. Why have a `refresh_state` method if you're not going to refresh the state?!